plugins/carbon_black_defense/komand_carbon_black_defense/connection/connection.py

```diff
@@ -1,35 +1,131 @@
-import komand
+import insightconnect_plugin_runtime
+from insightconnect_plugin_runtime.exceptions import PluginException
+
 from .schema import ConnectionSchema, Input
-from komand.exceptions import ConnectionTestException
+from insightconnect_plugin_runtime.exceptions import ConnectionTestException
 
 # Custom imports below
 import requests
+from typing import Optional
+import json
 
 
-class Connection(komand.Connection):
+class Connection(insightconnect_plugin_runtime.Connection):
     def __init__(self):
         super(self.__class__, self).__init__(input=ConnectionSchema())
 
-    def connect(self, params):
+        self.job_id = None
+
+    def connect(self, params=None):
+        if params is None:
+            params = {}
         self.logger.info("Connect: Connecting..")
         self.host = params.get(Input.URL)
         self.token = params.get(Input.API_KEY).get("secretKey")
+        self.org_key = params.get(Input.ORG_KEY)
         self.connector = params.get(Input.CONNECTOR)
+        self.headers = {"X-Auth-Token": f"{self.token}/{self.connector}"}
+
+    def get_job_id_for_enriched_event(self, process_name: str, event_id: str) -> Optional[str]:
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs",
+            json_data={
+                "criteria": {
+                    "process_name": [process_name],
+                },
+                "event_ids": event_id,
+            },
+        ).get("job_id")
+
+        self.logger.info(f"The response is {response}.")
+        self.logger.info(f"The url is: {self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs)")
+        if response:
+            job_id = response
+            return job_id
+        return None
+
+    def get_enriched_event_status(self, job_id: str = None):
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v1/orgs/{self.org_key}/enriched_events/search_jobs/{job_id}",
+            json_data={"cb_job_id": job_id},
+        )
+        self.logger.info(f"{response}")
+        self.logger.info(
+            f"The url is: {self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs/" f"{job_id}"
```

Comment:
Let's either remove these info logger lines, or save the formatted string to a variable to reference in both the `call_api()` invocation and the logger cal. Let's do the same for all instances of logging the URL.

Reply:
removed logger lines